### ISSUE TITLE
Validator: For contrasts, the number of weights should be equal the number of conditions

### ISSUE THREAD

**Issue by Remi-Gau** [bids-standard/stats-models#41]
The following should be invalid

```json
      "Contrasts": [
        {
          "Name": "incongruent_vs_congruent",
          "ConditionList": ["incongruent", "congruent"],
          "Weights": [1, -1, 1],
          "Test": "t"
        }
      ]
```

**Comment by effigies** [comment]
I don't know if we can impose this in general, since `x_*` would mean we don't know the number of conditions to check against. But in general I agree. Is it possible to do this in JSON validation or would it be purely in Python?

QQ: For F tests, I believe it's the case that the columns are supposed to match number of conditions, right? And is it the case that `nrows <= ncols`?

**Comment by Remi-Gau** [comment]
> I don't know if we can impose this in general, since `x_*` would mean we don't know the number of conditions to check against. 

true: to a certain extend the fine detail would have to be done on the implementation side, but in any case one would need to know what weight to give each condition so even for a globbing parttern there should something that will be applied for all conditions that would result from it.

> But in general I agree. Is it possible to do this in JSON validation or would it be purely in Python?

No idea.
 
> QQ: For F tests, I believe it's the case that the columns are supposed to match number of conditions, right? 

yup

> And is it the case that `nrows <= ncols`?

yup
